### PR TITLE
fix: enable testifylint on `staging/src/k8s.io/kubelet`

### PR DESCRIPTION
**Pull request by mmorel-35**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This fixes rules from testifylint on `staging/src/k8s.io/kubelet` package

### CODE REVIEW

**Comment by bart0sh** [comment]
@mmorel-35 Would it make sense to fix the rest of testifylint findings?
```
kubernetes (master) $ testifylint ./staging/src/k8s.io/kubelet/...
/home/ed/go/src/k8s.io/kubernetes/staging/src/k8s.io/kubelet/pkg/cri/streaming/request_cache_test.go:74:2: expected-actual: need to reverse actual and expected values
/home/ed/go/src/k8s.io/kubernetes/staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:351:3: go-require: require must only be used in the goroutine running the test function
/home/ed/go/src/k8s.io/kubernetes/staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:359:3: go-require: require must only be used in the goroutine running the test function
/home/ed/go/src/k8s.io/kubernetes/staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:467:2: require-error: for error assertions use require
/home/ed/go/src/k8s.io/kubernetes/staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:210:3: require-error: for error assertions use require
/home/ed/go/src/k8s.io/kubernetes/staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:219:3: require-error: for error assertions use require
/home/ed/go/src/k8s.io/kubernetes/staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:238:3: require-error: for error assertions use require
/home/ed/go/src/k8s.io/kubernetes/staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:240:3: require-error: for error assertions use require
/home/ed/go/src/k8s.io/kubernetes/staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:254:3: require-error: for error assertions use require
/home/ed/go/src/k8s.io/kubernetes/staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:256:3: require-error: for error assertions use require
/home/ed/go/src/k8s.io/kubernetes/staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:460:2: require-error: for error assertions use require
/home/ed/go/src/k8s.io/kubernetes/staging/src/k8s.io/kubelet/pkg/cri/streaming/request_cache_test.go:70:2: require-error: for error assertions use require
/home/ed/go/src/k8s.io/kubernetes/staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:52:2: require-error: for error assertions use require
/home/ed/go/src/k8s.io/kubernetes/staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:58:2: require-error: for error assertions use require
/home/ed/go/src/k8s.io/kubernetes/staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:69:2: require-error: for error assertions use require
/home/ed/go/src/k8s.io/kubernetes/staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:84:3: require-error: for error assertions use require
/home/ed/go/src/k8s.io/kubernetes/staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:93:3: require-error: for error assertions use require
/home/ed/go/src/k8s.io/kubernetes/staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:102:3: require-error: for error assertions use require
```

File: staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go [hunk 1]
```diff
@@ -457,13 +457,15 @@ func doClientStreams(t *testing.T, prefix string, stdin io.Writer, stdout, stder
 func readExpected(t *testing.T, streamName string, r io.Reader, expected string) {
 	result := make([]byte, len(expected))
 	_, err := io.ReadAtLeast(r, result, len(expected))
+	//nolint:testifylint  // always called in go func
```
Comment (bart0sh) [inline staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:460]: This doesn't seem to stop testifylint from complaining:
```
$ testifylint ./staging/src/k8s.io/kubelet/...
/home/ed/go/src/k8s.io/kubernetes/staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:461:2: require-error: for error assertions use require
/home/ed/go/src/k8s.io/kubernetes/staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:469:2: require-error: for error assertions use require
```

Comment (bart0sh) [reply staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:460]: nvm, figured this out. It's a golangci-lint instruction and golangci-lint seems to be happy:
```
$ ./hack/verify-golangci-lint.sh -n -a ./staging/src/k8s.io/kubelet/...
installing golangci-lint and logcheck plugin from hack/tools into config=/home/ed/go/src/k8s.io/kubernetes/hack/golangci-hints.yaml ./staging/src/k8s.io/kubelet/...
Congratulations! All files are passing lint :-)
```

**Comment by bart0sh** [comment]
/lgtm

/assign @mrunalp @derekwaynecarr @dchen1107 
for approval

File: staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go (hunk 1 above)
Comment (SergeyKanzhelev) [inline staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:460]: can you please clarify that those are not go func, but rather exec. As written, it is confusing as I cannot find the go func

Comment (SergeyKanzhelev) [reply staging/src/k8s.io/kubelet/pkg/cri/streaming/server_test.go:460]: /remove-lgtm

**Pull request closed without merging**